Mage.Sets/src/mage/cards/s/SunfrillImitator.java

```diff
@@ -88,16 +83,23 @@ class SunfrillImitatorEffect extends OneShotEffect {
     private static final CopyApplier copyApplier = new CopyApplier() {
         @Override
         public boolean apply(Game game, MageObject blueprint, Ability source, UUID targetObjectId) {
-            blueprint.setName("Sunfrill Imitator");
```

Comment:
Hm, I think this part might have been right before (the copy gets the name Sunfrill Imitator regardless of the name of the ability source). See implementations for Lazav the Multifarious or Sakashima the Impostor.

> 201.5. Text that refers to the [object](https://yawgatog.com/resources/magic-rules/#object) it's on by [name](https://yawgatog.com/resources/magic-rules/#name) means just that particular [object](https://yawgatog.com/resources/magic-rules/#object) and not any other [objects](https://yawgatog.com/resources/magic-rules/#object) with that [name](https://yawgatog.com/resources/magic-rules/#name), regardless of any [name](https://yawgatog.com/resources/magic-rules/#name) changes caused by game [effects](https://yawgatog.com/resources/magic-rules/#effect).

So clearly "Whenever Sunfrill Imitator attacks" is referring to the object by name, and thus `AttacksTriggeredAbility` uses the trigger phrase "Whenever {this} attacks, "... but I think "except its name is Sunfrill Imitator" is defining a characteristic and not referring to the source object.

(That said, I couldn't find an explicit reference in the card rulings or the Comprehensive Rules, and I'm not a judge.)